app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -60,21 +66,44 @@ public void afterTextChanged(Editable s) {
         vLoginButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Toast.makeText(getContext(), "Log In press", Toast.LENGTH_LONG).show();
-                //getPresenter().storeUsername(vUserNameInput.getText().toString());
+                if (TextUtils.isEmpty(vUserNameInput.getText())) {
+                    vUserNameInput.setError(getString(R.string.error_campo_incompleto));
+                } else if (!validarEmail(vUserNameInput.getText().toString())) {
+                    vUserNameInput.setError(getString(R.string.error_formato_invalido));
+                } else if (vPasswordInput.getText().toString().isEmpty()) {
+                    vPasswordInput.setError(getString(R.string.error_campo_incompleto));
+                } else {
+                    guardarUsuario(vUserNameInput.getText().toString());
+                    Toast.makeText(getContext(), "Log In press", Toast.LENGTH_LONG).show();
+                    //getPresenter().storeUsername(vUserNameInput.getText().toString());
+                }
             }
         });
 
         vSignUpButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                vUserNameInput.requestFocus();
                 Toast.makeText(getContext(), "Sign In press", Toast.LENGTH_LONG).show();
             }
         });
 
         vTermsAndConditions.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+    private void guardarUsuario(String userMail) {
```

Comment:
Change the name of the method and send the sharedPreference to the presenter.

Reply:
Done!